### PR TITLE
chore: Refactor create topic

### PR DESCRIPTION
**Pull request by suyash-naithani**
Signed-off-by: suyash-naithani <suyash.naithani@gmail.com>

**What this PR does / why we need it**:

> Export Create topic page and fix circular dependency
> Remove enum types for better performance
> Lift state for advance options switch
> Merge two different translation files for topic into one

### CODE REVIEW

File: .storybook/withMas.tsx [hunk 1]
```diff
@@ -32,7 +32,6 @@ export const withMas = (
                 datascienceoverview: () =>
                   import("../locales/en/datascienceoverview.json"),
                 kafkaoverview: () => import("../locales/en/kafkaoverview.json"),
-                topic: () => import("../locales/en/topic.json"),
```
Comment (HemaHG) [inline .storybook/withMas.tsx:35]: don't delete topic.json file, I am adding all the topic related code in topic.json file

Comment (suyash-naithani) [reply .storybook/withMas.tsx]: Reverted

File: src/Kafka/CreateKafkaTopic/CreateTopicPage.tsx [hunk 2]
```diff
@@ -13,6 +12,8 @@ export type CreateTopicPageProps = {
   checkTopicName: (value: string) => Promise<boolean>;
   availablePartitionLimit: number;
   constantValues: ConstantValues;
+  isSwitchChecked: boolean;
+  setIsSwitchChecked: (value: boolean) => void;
```
Comment (riccardo-forina) [inline src/Kafka/CreateKafkaTopic/CreateTopicPage.tsx:16]:
```suggestion
  onSwitchChange: (value: boolean) => void;
```

Comment (riccardo-forina) [reply src/Kafka/CreateKafkaTopic/CreateTopicPage.tsx:16]: Names should describe what is happening, not what should happen. That function is a callback for when the checkbox changes its value. The user of the component will decide how to handle that.

**Comment by hemahg** [comment]
the feature is already implemented in rhosak-ui, closing this in favor of rhosak-ui

**Pull request closed without merging**